scripts/core/exp-tensor-map.yml

```diff
@@ -0,0 +1,213 @@
+    - name: 64B
+      desc: "64B promotion type"
+    - name: 128B
+      desc: "128B promotion type"
+    - name: 256B
+      desc: "256B promotion type"
+
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Tensor map swizzle"
+name: $x_exp_tensor_map_swizzle_flags_t
+etors:
+    - name: NONE
+      desc: "No swizzle"
+    - name: 32B
+      desc: "32B swizzle"
+    - name: 64B
+      desc: "64B swizzle"
+    - name: 128B
+      desc: "128B swizzle"
+
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Tensor map OOB fill"
+name: $x_exp_tensor_map_oob_fill_flags_t
+etors:
+    - name: NONE
+      desc: "No OOB fill"
+    - name: REQUEST_ZERO_FMA
+      desc: "Refer to NVIDIA docs"
+
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Encode tensor map with image data"
+class: $xTensorMap
+name: EncodeIm2ColExp
+details:
+    - "Map encode using im2col."
+params:
+    - type: $x_device_handle_t
+      name: hDevice
+      desc: "[in] Handle of the device object."
+    - type: $x_exp_tensor_map_data_type_flags_t
+      name: TensorMapType
+      desc: "[in] Data type of the tensor object."
+    - type: uint32_t
+      name: TensorRank
+      desc: "[in] Dimensionality of tensor; must be at least 3."
+    - type: void*
+      name: GlobalAddress
+      desc: "[in] Starting address of memory region described by tensor."
+    - type: const uint64_t*
+      name: GlobalDim
+      desc: "[in] Array containing tensor size (number of elements) along each of the TensorRank dimensions."
+    - type: const uint64_t*
+      name: GlobalStrides
+      desc: "[in] Array containing stride size (in bytes) along each of the TensorRank - 1 dimensions."
+    - type: const int*
+      name: PixelBoxLowerCorner
+      desc: "[in] Array containing DHW dimensions of lower box corner."
+    - type: const int*
+      name: PixelBoxUpperCorner
+      desc: "[in] Array containing DHW dimensions of upper box corner."
+    - type: uint32_t
+      name: ChannelsPerPixel
+      desc: "[in] Number of channels per pixel."
+    - type: uint32_t
+      name: PixelsPerColumn
+      desc: "[in] Number of pixels per column."
+    - type: const uint32_t*
+      name: ElementStrides
+      desc: "[in] Array containing traversal stride in each of the TensorRank dimensions."
+    - type: $x_exp_tensor_map_interleave_flags_t
+      name: Interleave
+      desc: "[in] Type of interleaved layout the tensor addresses"
+    - type: $x_exp_tensor_map_swizzle_flags_t
+      name: Swizzle
+      desc: "[in] Bank swizzling pattern inside shared memory"
+    - type: $x_exp_tensor_map_l2_promotion_flags_t
+      name: L2Promotion
+      desc: "[in] L2 promotion size."
+    - type: $x_exp_tensor_map_oob_fill_flags_t
+      name: OobFill
+      desc: "[in] Indicates whether zero or special NaN constant will be used to fill out-of-bounds elements."
```

Comment:
This is a lot of arguments and its not extensible. I think some or all of these should move into a properties struct.

Reply:
That would make sense, however this extension is currently only to match the similar CUDA API, so I think it makes more sense to keep it close to the original one.

Currently no other hardware has a need for this, it may change in the future but without knowing the specifics of how other hardware will handle this we can't define a generic interface, so we'd like to have this as an experimental CUDA specific interface, to be revisited if and when other hardware also needs it.